frontend/src/static/js/v_summary.js

```diff
@@ -352,22 +366,31 @@ window.vSummary = {
     },
     splitCommitsWeek(user) {
       const { commits } = user;
-      const leng = commits.length;
 
       const res = [];
-      for (let weekId = 0; weekId < (leng - 1) / 7; weekId += 1) {
+
+      const sinceDate = dateRounding(this.filterSinceDate, 1);
+      const untilDate = this.filterUntilDate;
+
+      const sinceMs = (new Date(sinceDate)).getTime();
+      const untilMs = (new Date(untilDate)).getTime();
+
+      const diff = Math.round(Math.abs((untilMs - sinceMs) / DAY_IN_MS));
+
+      for (let weekId = 0; weekId < diff / 7; weekId += 1) {
+        const startOfWeekMs = sinceMs + (weekId * WEEK_IN_MS);
+
         const week = {
           insertions: 0,
           deletions: 0,
-          date: commits[weekId * 7].date,
+          date: getDateStr(startOfWeekMs),
         };
 
-        for (let dayId = 0; dayId < 7; dayId += 1) {
-          const commit = commits[(weekId * 7) + dayId];
-          if (commit) {
-            week.insertions += commit.insertions;
-            week.deletions += commit.deletions;
-          }
+        while (commits.length > 0
```

Comment:
add a comment to explain these a little. it might not be as intuitive as the previous version